components/enterprise/subscriptions.tf

```diff
@@ -1,62 +1,37 @@
-
-  subscriptions = merge(
-    local.cft,
-    local.cft_sandbox,
-    local.cft_non_production,
-    local.cft_production,
-
-    local.sds,
-    local.sds_sandbox,
-    local.sds_non_production,
-    local.sds_production,
-
-    local.crime,
-
-    local.heritage,
-    local.heritage_sandbox,
-    local.heritage_non_production,
-    local.heritage_production,
-
-    local.security,
-
-    local.platform,
-    local.platform_sandbox,
-    local.platform_non_production,
-    local.platform_production,
-  )
-}
-
 module "subscription" {
   for_each = local.subscriptions
 
-  source = "../../modules/subscription"
-  name   = each.key
-  value  = each.value
+  source      = "../../modules/subscription"
+  name        = each.key
+  value       = each.value
+  common_tags = module.tags[each.key].common_tags
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
 
   billing_account_name    = var.billing_account_name
   enrollment_account_name = var.enrollment_account_name
 }
+
+module "acme" {
+  for_each = { for k, v in local.subscriptions : k => v if try(v.deploy_acme, false) }
+
+  source = "git::https://github.com/hmcts/terraform-module-acme-function.git?ref=master"
+
+  location                       = var.location
+  env                            = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
+  common_tags                    = module.tags[each.key].common_tags
+  dns_contributor_group_id       = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].dns_contributor_group_id
+  product                        = var.product
+  resource_group_name            = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].resource_group_name
+  resource_group_id              = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].resource_group_id
+  asp_id                         = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].asp_id
+  subscription_id                = module.subscription[each.key].subscription_id.subscription_id
+  acme_storage_account_repl_type = try(each.value.acme_storage_account_repl_type, "ZRS")
+}
+
+module "tags" {
+  for_each    = local.subscriptions
+  source      = "git::https://github.com/hmcts/terraform-module-common-tags.git?ref=master"
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
+  product     = try(each.value.product, join("-", [regex("cft|sds", [each.value.group][0]), "platform"]), replace(regex("security", [each.value.group][0]), "/([A-Za-z]).*/", "soc"), replace([each.value.group][0], "/([A-Za-z])_.*/", "$1"))
```

Comment:
could the regex be extracted to a local so it's not repeated and it can be given a clear name explaining what it's doing?